doc/developer/design/00000000_template.md

```diff
@@ -1,42 +1,64 @@
-# <Insert Name of Feature>
+- Feature name: (Insert name of feature)
+- Associated issues: (Insert list of associated epics or issue)
+- Associated PRs: (List associated PRs)
 
-## Summary
+# Summary
+[summary]: #summary
 
-<!--
-// Brief, high-level overview. A few sentences long.
-// Be sure to capture the customer impact - framing this as a release note may be useful.
--->
+One paragraph to explain the feature.
 
-## Goals
+# Motivation
+[motivation]: #motivation
 
-<!--
-// Enumerate the concrete goals that are in scope for the project.
--->
+Why are we doing this? What use-cases does it enable? What is the desired outcome?
 
-## Non-Goals
+# Explanation
```

Comment:
Do I understand correctly that this section is a substantial addition compared with the previous template, while the other sections propose an improved organization of similar contents as previously? 

Nothing against having an explanation that is of the persuasive sort, arguing how the new reality could look like if the design is implemented. However, I'd just like to call out here that we might be not only revising the design review process, but also the requirements for a design document.

Reply:
Clarified the motivation. Thanks for pointing this out!